minio/mc
# feat: add put action

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

add put action support -s size -P part_numbers

- [x] avoid the so many call with one upload GetBucketObjectLockConfig/HeadBucket
- [x] avoid mc cp does with complex vairations( source is local path/ target is s3 )

## Motivation and Context


## How to test this PR?

1.deploy a minio
2. mc admin trace -a alias | grep s3.
3. set env MC_REGION=value
4.  `mc put object minio/bucket` , `mc put object minio/bucket/someobject` and `mc put object minio/bucket/dir/`

You will see just upload call here.

## Types of changes

## Checklist:
